FredBoat/src/main/java/fredboat/command/admin/SetAvatarCommand.java

```
@@ -0,0 +1,91 @@
+package fredboat.command.admin;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import fredboat.FredBoat;
+import fredboat.command.util.HelpCommand;
+import fredboat.commandmeta.abs.Command;
+import fredboat.commandmeta.abs.CommandContext;
+import fredboat.commandmeta.abs.ICommandRestricted;
+import fredboat.messaging.internal.Context;
+import fredboat.perms.PermissionLevel;
+import net.dv8tion.jda.core.entities.Icon;
+import net.dv8tion.jda.core.entities.Message.Attachment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * This command allows a bot admin to change the avatar of FredBoat
+ */
+public class SetAvatarCommand extends Command implements ICommandRestricted {
+
+    private static final Logger log = LoggerFactory.getLogger(SetAvatarCommand.class);
+
+    @Override
+    public void onInvoke(CommandContext context) {
+        String imageUrl = null;
+
+        if (!context.msg.getAttachments().isEmpty()) {
+            Attachment attachment = context.msg.getAttachments().get(0);
+            imageUrl = attachment.getUrl();
+        } else if (context.args.length > 1) {
+            imageUrl = context.args[1];
+        }
+
+        if (imageUrl == null) {
+            HelpCommand.sendFormattedCommandHelp(context);
+        } else {
+            if (imageUrl.startsWith("http://") || imageUrl.startsWith("https://")) {
+                String type = getImageMimeType(imageUrl);
+
+                if (type.equals("image/jpeg") || type.equals("image/png") || type.equals("image/gif") || type.equals("image/webp")) {
+                    setBotAvatar(context, imageUrl);
+                }
```

This whole if construct is still having several branches where it silently fails.

You probably should invert the flow, and return if setBotAvatar() is successful, otherwise have it fall through to send the help.